packages/core/stories/combo-box/combo-box.stories.tsx

```diff
@@ -702,3 +702,93 @@ export const FreeText: StoryFn<ComboBoxProps> = (args) => {
     </ComboBox>
   );
 };
+
+export const SelectAll: StoryFn<ComboBoxProps> = (args) => {
+  const [value, setValue] = useState(getTemplateDefaultValue(args));
+  const [selected, setSelected] = useState<string[]>([]);
+  const allSelectedOptionValue = "all";
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    // React 16 backwards compatibility
+    event.persist();
+
+    const value = event.target.value;
+    setValue(value);
+  };
+
+  const handleSelectionChange = (
+    event: SyntheticEvent,
+    newSelected: string[]
+  ) => {
+    let newOptionsSelected = [...newSelected];
+    //case: clear all if select all is unselected
+    if (
```

Comment:
The comment above says this check is for is all is unselected but the logic is for if all is selected and it was previously selected?

Reply:
corrected the comment, it for case where all select all was previously selected, if user unselects any option, then select all should be unselected

Reply:
Whenever we need comments  to explain code, it's usually a sign that the code does not explain itself and there is potential to refactor.

As these code are docs, it's worthwhile to consider how we could make the code, explain itself.

```
  const handleSelectionChange: DropdownProps["onSelectionChange"] = (
    event,
    newSelected
  ) => {
    let newOptionsSelected = [...newSelected];
    const wasAllSelected = selected.includes(allSelectedOptionValue);
    const isAllSelected = newOptionsSelected.includes(allSelectedOptionValue);

    if (wasAllSelected) {
      if (isAllSelected) {
        newOptionsSelected = newOptionsSelected.filter(
          (el) => el !== allSelectedOptionValue
        );
      } else {
        newOptionsSelected = [];
      }
    } else if (
      isAllSelected ||
      (!isAllSelected && newOptionsSelected.length === usStates.length)
    ) {
      newOptionsSelected = [...usStates, allSelectedOptionValue];
    }
    setSelected(newOptionsSelected);
    args.onSelectionChange?.(event, newOptionsSelected);
  };
```

How about this, any easier to read?

Reply:
yes its easier! I have updated this. Thanks Mark for this.